source/common/common/utility.h

```diff
@@ -50,30 +50,35 @@ class DateFormatter {
 private:
   std::string parse(const std::string& format_string);
 
-  typedef std::vector<int32_t> SubsecondOffsets;
-  std::string fromTimeAndPrepareSubsecondOffsets(time_t time,
-                                                 SubsecondOffsets& subsecond_offsets) const;
+  typedef std::vector<int32_t> SpecifierOffsets;
+  std::string fromTimeAndPrepareSpecifierOffsets(time_t time, SpecifierOffsets& specifier_offsets,
+                                                 const std::string& seconds_str) const;
 
-  // A container to hold a subsecond specifier (%f, %Nf) found in a format string.
-  struct SubsecondSpecifier {
-    SubsecondSpecifier(const size_t position, const size_t width, const std::string& segment)
-        : position_(position), width_(width), segment_(segment) {}
+  // A container to hold a specifiers (%f, %Nf, %s) found in a format string.
+  struct Specifier {
+    Specifier(const size_t position, const size_t width, const std::string& segment,
+              const bool second = false)
```

Comment:
Nit: It would be neater to have a different constructor for second and subsecond, since you can't have width with second.

Reply:
Updated in here b79ada9